packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -0,0 +1,373 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'color_scheme.dart';
+import 'colors.dart';
+import 'constants.dart';
+import 'ink_well.dart';
+import 'input_border.dart';
+import 'input_decorator.dart';
+import 'material.dart';
+import 'material_state.dart';
+import 'search_bar_theme.dart';
+import 'text_field.dart';
+import 'text_theme.dart';
+import 'theme.dart';
+
+/// A Material Design search bar.
+///
+/// Search bars include a [leading] Search icon, a text input field and optional
+/// [trailing] icons. A search bar is typically used to open a search view.
+/// It is the default trigger for a search view.
+///
+/// The [leading] widget is on the left side of the bar and should contain either
+/// a navigational action (such as a menu or up-arrow) or a non-functional
+/// search icon.
+///
+/// The [trailing] is an optional list. There should be up to two action icons
+/// that are located on the bar’s right-hand side. These actions can represent
+/// additional modes of searching (like voice search), a separate high-level
+/// action (such as current location) or an overflow menu.
+class SearchBar extends StatefulWidget {
+  /// Creates a Material Design search bar.
```

Comment:
I think this is all that you need to say for the constructor's API doc.

Reply:
Removed the rest!